pkg/providers/snow/defaults_test.go

```diff
@@ -4,35 +4,28 @@ import (
 	"errors"
 	"testing"
 
+	"github.com/golang/mock/gomock"
 	. "github.com/onsi/gomega"
 )
 
 func TestSetDefaultSshKey(t *testing.T) {
 	g := newConfigManagerTest(t)
 	g.machineConfig.Spec.SshKeyName = ""
-	wantKey := "eksa-default-cp-machine.pem"
-	wantVal := "pem val"
-	g.aws.EXPECT().EC2CreateKeyPair(g.ctx, wantKey).Return(wantVal, nil)
-	k, v, err := g.defaulters.GenerateDefaultSshKey(g.ctx, g.machineConfig)
+	g.aws.EXPECT().EC2ImportKeyPair(g.ctx, "eksa-default", gomock.Any()).Return(nil).Times(2)
+	err := g.defaulters.GenerateDefaultSshKey(g.ctx, g.machineConfig)
 	g.Expect(err).To(Succeed())
-	g.Expect(k).To(Equal(wantKey))
-	g.Expect(v).To(Equal(wantVal))
 }
 
 func TestSetDefaultSshKeySkip(t *testing.T) {
 	g := newConfigManagerTest(t)
-	k, v, err := g.defaulters.GenerateDefaultSshKey(g.ctx, g.machineConfig)
+	err := g.defaulters.GenerateDefaultSshKey(g.ctx, g.machineConfig)
 	g.Expect(err).To(Succeed())
-	g.Expect(k).To(Equal(""))
-	g.Expect(v).To(Equal(""))
 }
 
 func TestSetDefaultSshKeyError(t *testing.T) {
 	g := newConfigManagerTest(t)
 	g.machineConfig.Spec.SshKeyName = ""
-	g.aws.EXPECT().EC2CreateKeyPair(g.ctx, "eksa-default-cp-machine.pem").Return("v", errors.New("error"))
-	k, v, err := g.defaulters.GenerateDefaultSshKey(g.ctx, g.machineConfig)
+	g.aws.EXPECT().EC2ImportKeyPair(g.ctx, "eksa-default", gomock.Any()).Return(errors.New("error"))
```

Comment:
I wonder if we should inject a ssh key generator in the defaulter, so we don't need to use `Any()` (eg. we don't generate the real ssh keys in these unit tests). That would make this assertion a bit more robust and the test faster

Reply:
I did something similar in Tinkerbell. The driver was deterministic behavior for unit testing.

Reply:
updated